### PR TITLE
Adds --importfile to EasyCert command line tool

### PR DESCRIPTION
**Pull request by vertexmc**

### CODE REVIEW

File: synapse/lib/certdir.py [hunk 1]
```diff
@@ -579,6 +581,41 @@ def getUserKeyPath(self, name):
...
+    def importFile(self, path, mode, outp=None):
+        '''
+        Imports certs and keys into the Synapse cert directory
+
+        Args:
+            path (str): The path of the file to be imported.
+            mode (str): The certdir subdirectory to import the file into.
+
+        Examples:
+            importFile('mycoolca.crt', 'cas')
```
Comment (vEpiphyte) [inline synapse/lib/certdir.py]:
```
Examples:
       Import CA certifciate 'mycoolca.crt' to the 'cas' directory.

            certdir.importFile('mycoolca.crt', 'cas')
```

File: synapse/lib/certdir.py [hunk 2]
```diff
@@ -579,6 +581,41 @@ def getUserKeyPath(self, name):
...
+            None
+        '''
+        if not os.path.isfile(path):
+            raise s_common.NoSuchFile('File does not exist')
+
+        fname = os.path.split(path)[1]
+        parts = fname.split('.', 1)
+        ext = parts[1] if len(parts) is 2 else None
+
+        if not ext or ext not in('crt', 'key', 'p12'):
```
Comment (vEpiphyte) [inline synapse/lib/certdir.py]: ``in(`` should have a space

File: synapse/tests/test_lib_certdir.py [hunk 3]
```diff
@@ -405,3 +406,44 @@ def test_certdir_users_csr(self):
...
+    def test_certdir_importfile(self):
+        with self.getCertDir() as cdir:  # type: s_certdir.CertDir
+            with self.getTestDir() as testpath:
+
+                # File doesn't exist
+                fpath = s_common.genpath(testpath, 'not_real.crt')
+                self.raises(NoSuchFile, cdir.importFile, fpath, 'cas')
+
+                # File has unsupported extension
+                fpath = s_common.genpath(testpath, 'coolpic.bmp')
```
Comment (vEpiphyte) [inline synapse/tests/test_lib_certdir.py]: cool story bro

File: synapse/tests/test_lib_certdir.py [hunk 4]
```diff
@@ -405,3 +406,44 @@ def test_certdir_users_csr(self):
...
+                with s_common.genfile(fpath) as fd:
+                    self.raises(BadFileExt, cdir.importFile, fpath, 'cas')
+
+                tests = (
+                    ('cas', 'coolca.crt'),
```
Comment (vEpiphyte) [inline synapse/tests/test_lib_certdir.py]: add a .p12 file here for importing

Comment (vertexmc) [reply synapse/tests/test_lib_certdir.py]: added

**Pull request merged**